### ISSUE TITLE
Directory depth penalty

### ISSUE THREAD

**Issue by chamini2** [atom/fuzzaldrin#27]
I understand the scoring algorithm uses some sort of penalty to a file if it was too deep inside the directory structure, but sometimes this doesn't make sense, I'll post an example and explain:

![image](https://cloud.githubusercontent.com/assets/2745502/10895238/53946a88-8185-11e5-8a9d-f1c3d587081d.png)

As you can see, it now matches the probably expected file, but when we want to filter by saying *"it's in the `controllers/` directory"* 

<img width="508" alt="screen shot 2015-11-02 at 5 08 01 pm" src="https://cloud.githubusercontent.com/assets/2745502/10895295/a1463a2c-8185-11e5-90bd-88c646903c7b.png">

It penalizes the `application_controller` result because directory depth, or at least that's my guess. So why not start the deepness count of at the first directory match, in this case in the `controllers/` directory.

**Comment by 50Wliu** [comment]
Have you tried the new alternate scoring method present in Atom 1.1.0?  The new scoring method uses [fuzzaldrin-plus](https://github.com/jeancroy/fuzzaldrin-plus).

**Comment by jeancroy** [comment]
> So why not start the deepness count of at the first directory match, in this case in the controllers/ directory ?

From what I understand, directory dept penalty can be useful because user may not be aware of what file exist at very deep level. Additionally they may existed as vendor dependency (for example nested node_modules)

Deep path can trigger match length penalty but not necessarily. Let's take for example : `etc/lib/src/config/application_config.cfg`. Your proposal would then make that file as good as any file of the root.

I hope that cover the why.

-----

As for the what to do, since atom 1.1 fuzzy finder have an optional alternate scoring option which should cover this use case. (And 50Wliu beat me to it) 

Basically directory dept should not be a concern for a small difference. (like here a single directory)
If it's not behaving properly feel free to open an issue on the new project.

**Comment by chamini2** [comment]
This happens uses the alternate scoring option, thought I stated it in the original post.

But I do understand the reason @jeancroy, makes sense, thanks for the response.

**Comment by jeancroy** [comment]
Ok I'll give it another look, your example make sense to pass

**Comment by jeancroy** [comment]
Ok added a [test](https://github.com/jeancroy/fuzzaldrin-plus/commit/76fb30a5fbde17420822e069412c7b5e94030d88), your use-case works without any change on fuzzaldrin-plus@0.2.0 which adjust some weight vs what is currently in atom. 

Issue should probably be closed, if you have another issue please open it on https://github.com/jeancroy/fuzzaldrin-plus

**Comment by chamini2** [comment]
The test you added is missing the *failing* case:

     expect(bestMatch(candidates, 'con/appcon')).toBe candidates[1]

**Comment by jeancroy** [comment]
Ok sorry for that, I'm working on it. Could you open an issue in the proper repo ? (fuzzaldrin-plus, not classic fuzzaldrin)

In the meantime I believe `con appcon` should work for you.
Basically when the query have `n` slashes, I promote `n` last folder of path as if it where file name.
Then `config_appplication.rb` have 2 advantage: scored as a filename-only (no path before) & shorter ( length of entry is more critical for file-name)

I tried some things but it's not obvious to make that one pass without making something else fail. 
Maybe i'll just disable the feature that make a difference between `con appcon`  and `con/appcon`
